src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFs.java

```diff
@@ -363,6 +404,107 @@ private VariantContext removeNonRefAlleles(final VariantContext vc) {
         }
     }
 
+    /**
+     * Re-genotype (and re-annotate) a combined genomic VC
+     * @return a new VariantContext or null if the site turned monomorphic and we don't want such sites
+     */
+    private VariantContext regenotypeSomaticVC(final VariantContext originalVC, final ReferenceContext ref, final FeatureContext features, boolean includeNonVariants) {
+        Utils.nonNull(originalVC);
+
+        final VariantContext result;
+        if ( originalVC.isVariant()  && originalVC.getAttributeAsInt(VCFConstants.DEPTH_KEY,0) > 0 ) {
+            result = callSomaticGenotypes(originalVC);
+        } else if (includeNonVariants) {
+            result = originalVC;
+        }
+        else {
+            result = null;
+        }
+        return result;
+    }
+
+
+    /**
+     * Drop low quality alleles and call genotypes
+     * CombineGVCFs will convert calls to no-call (of varying ploidy, as is the case in somatic)
+     *
+     * @param vc input VariantContext with no-called genotypes
+     * @return a VC with called genotypes and low quality alleles removed, may be null
+     */
+    private VariantContext callSomaticGenotypes(final VariantContext vc) {
+        final VariantContextBuilder builder = new VariantContextBuilder(vc);
+        GenotypeBuilder gb;
+        final List<Genotype> newGenotypes = new ArrayList<>();
+        final GenotypesContext genotypes = vc.getGenotypes();
+        final double[] perAlleleLikelihoodSums = new double[vc.getAlleles().size()];  //needs the ref for the subsetting utils
+
+        for(final Genotype g : genotypes) {
+            gb = new GenotypeBuilder(g);
+            final double[] tlodArray = GATKProtectedVariantContextUtils.getAttributeAsDoubleArray(g, GATKVCFConstants.TUMOR_LOD_KEY, () -> null, 0.0);
+            final double[] variantAFArray = GATKProtectedVariantContextUtils.getAttributeAsDoubleArray(g, GATKVCFConstants.ALLELE_FRACTION_KEY, () -> null, 0.0);
+            double variantAFtotal = 0;
+            final List<Allele> calledAlleles = new ArrayList<>();
+            for(int i = 0; i < vc.getAlleles().size()-1; i++) {
+                variantAFtotal += variantAFArray[i];
+                if (tlodArray[i] > tlodThreshold) {
+                    calledAlleles.add(vc.getAlternateAllele(i));
+                    perAlleleLikelihoodSums[i+1] += tlodArray[i];
+                }
+            }
+            //hack for weird Mutect2 ploidy -- if the variant is non-homoplasmic, call the reference allele too
+            if(variantAFtotal < 1-afTolerance && (g.hasAD() ? g.getAD()[0] > 0 : true)) {
```

Comment:
I'm actually going to vote against the ternary in favor of `&& !(g.hasAD() && g.getAD()[0] == 0)`